reaktive/src/commonTest/kotlin/com/badoo/reaktive/observable/TakeObservableTests.kt

```diff
@@ -0,0 +1,109 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.test.base.assertDisposed
+import com.badoo.reaktive.test.base.assertError
+import com.badoo.reaktive.test.base.assertNotError
+import com.badoo.reaktive.test.observable.*
+import kotlin.test.Test
+import kotlin.test.assertFailsWith
+
+class TakeObservableTests :
+    ObservableToObservableTests by ObservableToObservableTests<Unit>({ take(1) }) {
+
+    private val upstream = TestObservable<Int?>()
+    val observer = upstream.take(10).test()
+
+    @Test
+    fun completes_WHEN_limit_of_1_is_reached() {
+        val observer = upstream.take(1).test()
+
+        upstream.onNext(42)
+
+        observer.assertComplete()
+        observer.assertNotError()
+    }
+
+    @Test
+    fun completes_WHEN_limit_of_2_is_reached() {
+        val observer = upstream.take(2).test()
+
+        upstream.onNext(42)
+        observer.assertNotComplete()
```

Comment:
Assertion is out of scope of the test, as you are verifying "completes when limit of 2 is reached". So produce two values and verify that completed.

Reply:
And add another test to verify that "does not complete when limit is not reached".

Reply:
Good idea.